CPPCheckPlugin/CPPCheckPluginPackage.cs

```diff
@@ -610,15 +610,29 @@ private static bool isCppFile(dynamic file)
 				// Using Reflection to get to the interface and access the property directly instead.
 				Type fileObjectType = file.GetType();
 				var vcFileInterface = fileObjectType.GetInterface("Microsoft.VisualStudio.VCProjectEngine.VCFile");
-				var fileTypeValue = vcFileInterface.GetProperty("FileType").GetValue((object)file);
-				Type fileTypeEnumType = fileTypeValue.GetType();
-				Debug.Assert(fileTypeEnumType.FullName == "Microsoft.VisualStudio.VCProjectEngine.eFileType");
-				var fileTypeEnumValue = Enum.GetName(fileTypeEnumType, fileTypeValue);
-				var fileTypeCppCodeConstant = "eFileTypeCppCode";
-				// First check the enum contains the value we're looking for
-				Debug.Assert(Enum.GetNames(fileTypeEnumType).Contains(fileTypeCppCodeConstant));
-				if (fileTypeEnumValue == fileTypeCppCodeConstant)
-					return true;
+                if (vcFileInterface == null)
+                {
+                    // We failed to get the interface so we can't lookup the real file type, and are left
+                    // with the option of hard-coding the value of eFileTypeCppCode, hoping that it won't
+                    // be changed in future versions.
+                    const int eFileTypeCppCode = 0;
+                    if (file.FileType == eFileTypeCppCode)

```

Comment:
This will not work if FileType is explicitly implemented.

Reply:
Oh well. Sadly, I don't understand these intricacies of `dynamic`. I tried getting rid of it altogether and am now stuck with some basic VS SDK functionality not working at all for some reason, not sure if I should revert all my refactoring at this point or try to solve the problem and finish it...

Reply:
The dynamic is there to allow to use the plugin with a variety of Visual Studio versions.

Reply:
I remember that, I was willing to sacrifice that compatibility in favor of maintainability.